src/server/lib/okta/register.ts

```diff
@@ -61,74 +62,183 @@ const sendRegistrationEmailByUserState = async ({
+				 *    token through Gateway
+				 *    And my status should remain PROVISIONED
+				 */
+				const tokenResponse = await reactivateUser(user.id);
+				if (!tokenResponse?.token.length) {
+					throw new OktaError({
+						message: `Okta user reactivation failed: missing re-activation token`,
+					});
+				}
+				const emailIsSent = await sendAccountWithoutPasswordExistsEmail({
+					to: user.profile.email,
+					activationToken: await encryptOktaRecoveryToken({
+						token: tokenResponse.token,
+						appClientId,
+						request_id,
+					}),
+					ref,
+					refViewId,
 				});
-			}
-			const emailIsSent = await sendAccountWithoutPasswordExistsEmail({
-				to: user.profile.email,
-				activationToken: await encryptOktaRecoveryToken({
-					token: tokenResponse.token,
-					appClientId,
-					request_id,
-				}),
-				ref,
-				refViewId,
-			});
-			if (!emailIsSent) {
-				trackMetric(
-					emailSendMetric('OktaAccountExistsWithoutPassword', 'Failure'),
+				if (!emailIsSent) {
+					trackMetric(
+						emailSendMetric('OktaAccountExistsWithoutPassword', 'Failure'),
+					);
+					throw new OktaError({
+						message: `Okta user reactivation failed: Failed to send email`,
+					});
+				}
+			} catch (error) {
+				// these users are in a special state where they used a passcode to create their account
+				// but failed to complete the flow by setting a password, leaving them in an unusual state
+				// where they're in the PROVISIONED state but cannot recover normally
+				// to remediate these users we have to deactivate them
+				// and then reactivate them in order to send them a create password email
+				if (error instanceof OktaError && error.code === 'E0000038') {
+					trackMetric(
+						'PasscodePasswordNotCompleteRemediation-Register-PROVISIONED-Start',
+					);
+
+					// 1. deactivate the user
+					try {
+						await deactivateUser(user.id);
+						trackMetric('OktaDeactivateUser::Success');
+					} catch (error) {
+						trackMetric('OktaDeactivateUser::Failure');
+						logger.error(
+							'Okta user deactivation failed',
+							error instanceof OktaError ? error.message : error,
+							{
+								request_id,
+							},
+						);
+						throw error;
+					}
+
+					trackMetric(
+						'PasscodePasswordNotCompleteRemediation-Register-PROVISIONED-Complete',
+					);
+
+					// rerun the sendRegistrationEmailByUserState function to catch the user in the DEPROVISIONED state
+					return sendRegistrationEmailByUserState({
+						email,
+						appClientId,
+						request_id,
+						ref,
+						refViewId,
+					});
+				}
+
+				logger.error(
+					'Okta user reactivation failed',
+					error instanceof OktaError ? error.message : error,
+					{
+						request_id,
+					},
 				);
-				throw new OktaError({
-					message: `Okta user reactivation failed: Failed to send email`,
-				});
+
+				// otherwise throw the error to the outer catch block
+				// as it's not handled in the if statement above
+				throw error;
```

Comment:
It seems like there's a lot of code dupilcation.  I think it makes it harder to read and maintain, but I won't hold a needed fix for that